src/GenesisMakeFileWindow.cpp

```diff
@@ -137,21 +158,67 @@ void GenesisMakeDirWindow::MessageReceived(BMessage* message)
 }
 
 ////////////////////////////////////////////////////////////////////////
-bool GenesisMakeDirWindow::CreateFolder(const char *dirpath, const char *dirname)
+bool GenesisMakeFileWindow::CreateFolder(const char *dirpath, const char *dirname)
 ////////////////////////////////////////////////////////////////////////
 {
 	BString dir;
 	dir.SetTo(dirpath);
 	dir+="/";
 	dir+=dirname;
 
-	if (create_directory(dir.String(), 0777)==B_OK)		// TODO: jo a 0777?
+	BEntry *entry = new BEntry(dir);
+	if (entry->Exists() && entry->IsDirectory())
+	{
+		BAlert *alert = new BAlert("Cannot create new folder","Folder already exists","OK",NULL,NULL,B_WIDTH_AS_USUAL,B_INFO_ALERT);
+		alert->Go();
 		return 1;
-	else
-		return 0;
+	}
+	
+	status_t status = create_directory(dir.String(), 0777);
+	if (status == B_OK)
+		return 1;
+		
+	BString errormsg;
+	switch (status){
+		case B_NOT_A_DIRECTORY:
+			errormsg.SetToFormat("Entry '%s' already exists", dirname);
+			break;
+		default:
+			errormsg.SetTo("Unknown error");
+			break;
+	}
+
+	BAlert *alert = new BAlert("Cannot create new folder",errormsg.String(),"OK",NULL,NULL,B_WIDTH_AS_USUAL,B_WARNING_ALERT);
+	alert->Go();
+
+	return 0;
 }
 
+////////////////////////////////////////////////////////////////////////
+bool GenesisMakeFileWindow::CreateFile(const char *path, const char *filename)
+////////////////////////////////////////////////////////////////////////
+{
+	BDirectory *dir = new BDirectory(path);
 
+	status_t status = dir->CreateFile(filename, NULL, true);
+	if (status == B_OK)
+		return 1;
+	
+	BString errormsg;
+	switch (status){
+		case B_FILE_EXISTS:
+			errormsg.SetToFormat("File '%s' already exists", filename);
```

Comment:
You may want to check if it's a folder here as well, and show a different message if it is.